crates/aptos-crypto/src/noise.rs

```diff
@@ -382,25 +377,21 @@ impl NoiseConfig {
 
         // <- payload
         let offset = cursor.position() as usize;
-        let received_encrypted_payload = &cursor.into_inner()[offset..];
 
-        let aead = Aes256Gcm::new(GenericArray::from_slice(&k));
-
-        let nonce = GenericArray::from_slice(&[0u8; AES_NONCE_SIZE]);
-        let ct_and_ad = Payload {
-            msg: received_encrypted_payload,
-            aad: &h,
-        };
-        let received_payload = aead
-            .decrypt(nonce, ct_and_ad)
-            .map_err(|_| NoiseError::Decrypt)?;
+        let aead = LessSafeKey::new(UnboundKey::new(&aead::AES_256_GCM, &k[..]).unwrap());
+        let mut in_out = cursor.into_inner()[offset..].to_vec();
+        let nonce = aead::Nonce::assume_unique_for_key([0u8; AES_NONCE_SIZE]);
+        let ptxt = aead.open_in_place(nonce, Aad::from(&h), &mut in_out)
```

Comment:
what is ptxt?

Reply:
ptxt = plaintext! I'll just rename it, if that's clearer?